### ISSUE TITLE
Delete user assignment before deleting a project

### ISSUE THREAD

**Issue by jaragunde** [Igalia/phpreport#172]
[Trac import]
Reported by: jaragunde
Original date: Wednesday, 29 February 2012 14:31

To delete a project, it can't be referenced in other tables. It means that:
- The project can't have assigned tasks.
- The project can't have assigned users or customers.

When you try to delete a project and you can't, the error message doesn't help to know why, and it's worse in the case you have already checked there are no assigned hours.

It's normal that a project with assigned task can't be deleted, because it could have a heavy impact in the company numbers, but user and customer assignations could be deleted with no problem, they aren't necessary if the project doesn't exist and their removal has no impact.

The problem with user feedback is global for all the application, so in this ticket we could focus on the cascade delete of the relations with users and customers.

**Comment by jaragunde** [comment]
[Trac import]
Comment by: jaragunde
Original date: Wednesday, 29 February 2012 14:36

Replying to [ticket:172 jaragunde]:
> 
> The problem with user feedback is global for all the application, so in this ticket we could focus on the cascade delete of the relations with users and customers.

The problem with feedback on error is being tracked in ticket #173.

**Comment by jaragunde** [comment]
There isn't a table assigning customers to projects any more, so this ticket is about removing user assignments before deleting the project. Updating the title with a more descriptive name.

**Comment by jaragunde** [comment]
Fixed in PR #614:
```
f0ae97d8 Delete unused test code. [Jacobo Aragunde Pérez]
3145b3b3 [#172] Delete user assignment when deleting a project. [Jacobo Aragunde Pérez]
55d87e67 [#496,#513] Return more specific errors on project deletion. [Jacobo Aragunde Pérez]
1b1aa84c Remove unused constructor for ProjectDAO. [Jacobo Aragunde Pérez]
```
